### PR TITLE
Audio: Mixer: Use frame aligned API

### PR DESCRIPTION
**Pull request by andrula-song**
Using frame aligned API to reduce the division calculation in mixer process function.

Signed-off-by: Andrula Song <andrula.song@intel.com>

### CODE REVIEW

File: src/audio/mixer/mixer.c [hunk 1]
```diff
@@ -179,6 +179,32 @@ static int mixer_reset(struct processing_module *mod)
...
 
+/* init and calculate the aligned setting for available frames and free frames retrieve*/
+static inline void mixer_set_frame_alignment(struct audio_stream __sparse_cache *source)
+{
+#if XCHAL_HAVE_HIFI3 || XCHAL_HAVE_HIFI4
+
+	/* Xtensa intrinsics ask for 8-byte aligned. 5.1 format SSE audio
+	 * requires 16-byte aligned.
+	 */
+	const uint32_t byte_align = source->channels == 6 ? 16 : 8;
```
Comment (lgirdwood) [inline src/audio/mixer/mixer.c:190]: @singalsu @dbaluta @andrula-song we need to come up with some arch SIMD utility macros/inline for this so we can use on all processing.

**Review by lgirdwood — COMMENTED** [review]
@andrula-song what's the MCPS gain here ?

File: src/audio/mixer/mixer_hifi3.c [hunk 2]
```diff
@@ -43,17 +39,15 @@ static void mix_n_s16(struct comp_dev *dev, struct audio_stream __sparse_cache *
...
 		m = n >> 2;
-		left = n & 0x03;
 
 		for (i = 0; i < m; i++) {
 			val1 = AE_ZERO32();
 			val2 = AE_ZERO32();
 			for (j = 0; j < num_sources; j++) {
 				/* load four 16 bit samples */
-				AE_LA16X4_IP(sample, inu[j], in[j]);
+				AE_L16X4_IP(sample, in[j], 8);
```
Comment (iuliana-prodan) [inline src/audio/mixer/mixer_hifi3.c:50]: Why do you add 8 at the base address?
Can you, please, add a comment to explain this?

Comment (andrula-song) [reply src/audio/mixer/mixer_hifi3.c:50]: Done

File: src/audio/mixer/mixer.c (hunk 1 above)
Comment (andrula-song) [reply src/audio/mixer/mixer.c:190]: would #define XTENSA_BYTE_ALIGN(channels) (channels) == 6 ? 16 : 8 in  numbers.h okay?

Comment (lgirdwood) [reply src/audio/mixer/mixer.c:190]: We should have an architecture generic macro in numbers.h e.g.
```c
#if ARCH_XTENSA
#if HIFI3
#define ARCH_SIMD_ALIGN(chan) 
#elif HIFI4
#define ARCH_SIMD_ALIGN(chan) 
#endif
#elif ARCH_other
#define ARCH_SIMD_ALIGN(chan)
#endif
```

Comment (andrula-song) [reply src/audio/mixer/mixer.c:190]: ok, I will create a new PR to add this macro and replace all such usage in our SOF.

**Comment by andrula-song** [comment]
the internal CI failed is not caused by my PR

File: src/audio/mixer/mixer.c (hunk 1 above)
Comment (lgirdwood) [reply src/audio/mixer/mixer.c:190]: ok, lets merge the new PR first and then redo this PR to use it.

Comment (andrula-song) [reply src/audio/mixer/mixer.c:190]: and the xtensa test result is below, the functions with '_1' is the new one, can improve about 15% cycles than the original ones.
Screenshot from 2023-02-02 14-43-29

**Review by kv2019i — APPROVED** [review]
Thanks, looks good!

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
SOFCI TEST

**Comment by lgirdwood** [comment]
@andrula-song can you check results here. Thanks !

**Comment by andrula-song** [comment]
SOFCI TEST

**Comment by andrula-song** [comment]
known issue SOF#6864

**Comment by lgirdwood** [comment]
> known issue SOF#6864

@andrula-song pls also check internal CI, this is the merge blocker.

**Comment by andrula-song** [comment]
hello @wszypelt , could you please help to trigger the CI test? It is a KD failed case, and the latest test of this PR is 02/03/2023.

**Comment by andrula-song** [comment]
now the CI tests are passed, the KD issue is not caused by this PR.

**Comment by kv2019i** [comment]
One known fail in https://sof-ci.01.org/sofpr/PR7029/build3792/devicetest/index.html , otherwise good. Proceeding with merge.

**Pull request merged**